core/codegen/src/derive/from_form.rs

```diff
@@ -40,7 +40,7 @@ fn validate_struct(gen: &DeriveGenerator, data: Struct<'_>) -> Result<()> {
 
     let mut names = ::std::collections::HashMap::new();
     for field in data.fields().iter() {
-        let id = field.ident.as_ref().expect("named field");
+        let id = crate::Ident::from(field.ident.as_ref().expect("named field").clone());
```

Comment:
Ideally we can write:

```suggestion
        let id: crate::Ident = field.ident.as_ref().expect("named field").into();
```